coldfront/core/project/views.py

```diff
@@ -902,15 +912,33 @@ def post(self, request, *args, **kwargs):
 
             project_user_update_form = ProjectUserUpdateForm(request.POST,
                                                              initial={'role': project_user_obj.role.name,
-                                                                      'enable_notifications': project_user_obj.enable_notifications}
-                                                             )
+                                                                      'enable_notifications': project_user_obj.enable_notifications})
 
             if project_user_update_form.is_valid():
                 form_data = project_user_update_form.cleaned_data
-                project_user_obj.enable_notifications = form_data.get(
-                    'enable_notifications')
-                project_user_obj.role = ProjectUserRoleChoice.objects.get(
-                    name=form_data.get('role'))
+                project_user_obj.enable_notifications = form_data.get('enable_notifications')
+
+                old_role = project_user_obj.role
+                new_role = ProjectUserRoleChoice.objects.get(name=form_data.get('role'))
+                only_manager = not project_obj.projectuser_set.filter(~Q(pk=project_user_pk),
+                                                                      role__name__in=['Manager']).exists()
+
+                # the only manager of this project, is demoting himself
```

Comment:
It could also be that a PI is changing the Manager's role, or that Manager B is changing Manager A's role. (I.e., it's not necessarily the case that a manager is operating on him/herself.)

We also need to handle the case when a Manager is being promoted to PI.

Reply:
yeah, if other managers exist, one can freely demote. If only _manager is true, it means that this can either be the only manager, or another PI. What happens when a PI demotes a manager, and himself doesn't have active notifications? I think the simplest would be to have the strict rule that all PIs get active notifications when the last manager is removed / removes himself.

Reply:
Okay, from the meeting, we've decided: when the last manager demotes him/herself to a regular user, all notifications for PIs will be re-activated.

The case where the current logic breaks:
- `old_role` is "Manager"
- `new_role` is "Principal Investigator" (i.e., promotion, not demotion)

If there are no PIs yet, we enter the case at line 930. But the manager is attempting to become a PI, so getting the error message ('The project must have at least one PI or manager with notifications enabled.') seems unexpected. The manager should be able to become a PI.

It there are PIs already, we enter the `else` case, wherein we re-enable all notifications for all PIs. But we needn't do that, because this manager is becoming a PI, and he still has notifications enabled.

Reply:
@ViRu-ThE-ViRuS I spoke with @kmuriki today, and it looks like we want to disable upgrading the role to "Principal Investigator" entirely, so this case is no longer relevant. Can you remove that choice from the dropdown?

Reply:
@matthew-li okay! Sounds good

Reply:
Yes, let's limit the options in these two places to just "Manager" and "User".

Reply:
thanks!